### PR TITLE
Protractor test integration

### PR DESCRIPTION
**Pull request by chalettu**
This pull request adds support for being able to run protractor tests on the project.

### CODE REVIEW

**Comment by chalettu** [comment]
@miq-bot add_label enhancement

File: client/app/states/dashboard/dashboard.e2e.js [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    browser.get(protractorConfig.baseUrl);
+  });
+  it('should have two counters up top', function() {
+    var list = element.all(by.css('.ss-dashboard__card-primary__count'));
+    expect(list.count()).toBe(2);
+  });
+
+  it('should have counters have a very specific non zero numnber', function() {
+    var counters = element.all(by.css('.ss-dashboard__card-primary__count h2'));
+     // var text = counters.get(1).getText()
```
Comment (AllenBW) [inline client/app/states/dashboard/dashboard.e2e.js]: commented text!

File: client/app/states/dashboard/dashboard.e2e.js [hunk 2]
```diff
@@ -0,0 +1,20 @@
+/* eslint angular/log: 0 */
+/* eslint no-console: 0 */
+/* eslint no-undef: 0 */
```
Comment (AllenBW) [inline client/app/states/dashboard/dashboard.e2e.js]: hmmm i wonder if adding an `.eslintignore` to root with `**/*.e2e.js` would be better than specifying all the rules to ignore 🤔

File: client/app/states/login/login.e2e.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
+/* eslint angular/log: 0 */
+/* eslint no-console: 0 */
+
+describe('pages with login', function() {
+  it('should log in with a non-Angular page', function() {
+    browser.sleep(10000);
+   //       browser.driver.executeScript('return window.sessionStorage.getItem("ngStorage-token");').then(function(retValue) {
```
Comment (AllenBW) [inline client/app/states/login/login.e2e.js]: commented text!

File: package.json [hunk 4]
```diff
@@ -97,6 +98,7 @@
     "optimize-css-assets-webpack-plugin": "1.3.0",
     "phantomjs-prebuilt": "2.1.14",
     "postcss-loader": "1.2.1",
+    "protractor": "^4.0.14",
```
Comment (AllenBW) [inline package.json]: keeping with observing absolute versions...  (kill da 🥕 )

File: package.json [hunk 5]
```diff
@@ -14,6 +14,7 @@
     "build": "webpack --bail --progress --profile --config config/webpack.prod.js",
     "start": "webpack-dev-server --open --progress --config config/webpack.dev.js",
     "test": "yarn vet && karma start karma.conf.js --single-run",
+    "test-e2e": "./node_modules/.bin/protractor protractor.conf.js",
```
Comment (AllenBW) [inline package.json]: can we rename this `test:e2e`?  keeping with `test:watch`

File: docs/developer_setup.md [hunk 6]
```diff
@@ -41,3 +41,17 @@ http://localhost:3000 order and serve up the REST API.
...
+```yarn test ```  
+
+If you would like to run the integration tests for the project please ensure you have selenium server installed and running  
+
+- ``` yarn global add protractor ```
+- ``` webdriver-manager update ```  
+- ``` webdriver-manager start```
+
+Now that your selenium server is running and listening open another terminal window and from the projects root directory, run   
+```yarn test-e2e ```
```
Comment (AllenBW) [inline docs/developer_setup.md]: can we rename this test:e2e? keeping with test:watch

File: docs/developer_setup.md [hunk 7]
```diff
@@ -41,3 +41,17 @@ http://localhost:3000 order and serve up the REST API.
   the service UI, which will initiate the UI listening on _http://localhost:3001_, and talking to the REST API at
   _http://[::1]:3000_.  This command will also open a browser page to  _http://localhost:3001/login_.
   - `yarn start`
+
+### Testing
```
Comment (AllenBW) [inline docs/developer_setup.md]: There is already a section called testing in `contributing.md`, this section might make more sense there

File: client/app/states/login/login.html [hunk 8]
```diff
@@ -34,7 +34,7 @@ <h1 id="brand" ng-bind-html="vm.text.brand"></h1><!--/#brand-->
             </div>
 
             <div class="col-xs-4 col-sm-4 col-md-4 pull-right submit">
-              <button type="submit" class="btn btn-primary btn-lg" tabindex="4" translate>Log In</button>
+              <button type="submit" class="btn btn-primary btn-lg" id="loginButton" tabindex="4" translate>Log In</button>
```
Comment (AllenBW) [inline client/app/states/login/login.html]: looks like the id was added just for the test, but it servers no other purpose, in efforts to keep it clean/un-confusing any way you can find the button by the type or innerHtml? (`submit` is unique to this element)

File: client/app/states/dashboard/dashboard.e2e.js (hunk 2 above)
Comment (chalettu) [inline client/app/states/dashboard/dashboard.e2e.js]: Thats definitely an option.  I was wondering if there was a cleaner way to deal with this.

**Comment by chriskacerguis** [comment]
@chalettu can you please add your PT story, and update your card?

**Comment by chriskacerguis** [comment]
PT Story: https://www.pivotaltracker.com/story/show/137095587

**Pull request merged**